ui/app/models/secret-engine.js

```diff
@@ -22,68 +22,121 @@ const validations = {
     possibleValues: [2, 1],
     // This shouldn't be defaultValue because if no version comes back from API we should assume it's v1
     defaultFormValue: 2, // Set the form to 2 by default
-  }),
-  config: belongsTo('mount-config', { async: false, inverse: null }),
-  local: attr('boolean', {
-    helpText:
-      'When Replication is enabled, a local mount will not be replicated across clusters. This can only be specified at mount time.',
-  }),
-  sealWrap: attr('boolean', {
-    helpText:
-      'When enabled - if a seal supporting seal wrapping is specified in the configuration, all critical security parameters (CSPs) in this backend will be seal wrapped. (For K/V mounts, all values will be seal wrapped.) This can only be specified at mount time.',
-  }),
+  })
+  version;
+
+  // SSH specific attributes
+  @attr('string') privateKey;
+  @attr('string') publicKey;
+  @attr('boolean', {
+    defaultValue: true,
+  })
+  generateSigningKey;
+
+  // AWS specific attributes
+  @attr('string') lease;
+  @attr('string') leaseMax;
+
+  // Returned from API response
+  @attr('string') accessor;
+
   // KV 2 additional config default options
-  maxVersions: attr('number', {
+  @attr('number', {
     defaultValue: 0,
     label: 'Maximum number of versions',
     subText:
       'The number of versions to keep per key. Once the number of keys exceeds the maximum number set here, the oldest version will be permanently deleted. This value applies to all keys, but a key’s metadata settings can overwrite this value. When 0 is used or the value is unset, Vault will keep 10 versions.',
-  }),
-  casRequired: attr('boolean', {
+  })
+  maxVersions;
+  @attr('boolean', {
     defaultValue: false,
     label: 'Require Check and Set',
     subText:
       'If checked, all keys will require the cas parameter to be set on all write requests. A key’s metadata settings can overwrite this value.',
-  }),
-  deleteVersionAfter: attr({
+  })
+  casRequired;
+  @attr({
     defaultValue: 0,
     editType: 'ttl',
     label: 'Automate secret deletion',
     helperTextDisabled: 'A secret’s version must be manually deleted.',
     helperTextEnabled: 'Delete all new versions of this secret after',
-  }),
+  })
+  deleteVersionAfter;
+
+  /* GETTERS */
+  get modelTypeForKV() {
+    const engineType = this.engineType;
+    if ((engineType === 'kv' || engineType === 'generic') && this.version === 2) {
+      return 'secret-v2';
+    }
+    return 'secret';
+  }
+  get isV2KV() {
+    return this.modelTypeForKV === 'secret-v2';
+  }
 
-  modelTypeForKV: computed('engineType', 'version', function () {
-    const type = this.engineType;
-    let modelType = 'secret';
-    if ((type === 'kv' || type === 'generic') && this.version === 2) {
-      modelType = 'secret-v2';
+  get attrs() {
+    return expandAttributeMeta(this, this.formFields);
+  }
+
+  get fieldGroups() {
+    return fieldToAttrs(this, this.formFieldGroups);
```

Comment:
If I remember correctly - these don't play nicely with glimmer components because of how they update. It make forms really buggy. So we need to use the model decorator or catch the initial result of expandAttributeMeta 

If you look at `models/oidc/client.js` that's how we had to do it before the decorator existed.

Reply:
Thanks, I forgot about that! Will update 👍

Reply:
I can't find the original conversation - but there is some small context here https://github.com/hashicorp/vault/pull/18229#discussion_r1040218216

Reply:
It's probably be worth adding an assertion as a reminder. It's not easy to remember - and I can't find any record of the original issue 🤯

Reply:
For posterity: ran into an issue when doing it the model-caching way like in `models/oidc/client`-- the groups don't update after the first time the groups are calculated, which results in values that should only be shown for certain secret engines being shown for all subsequent ones (assuming you choose that one first). Vice versa is also true -- if you choose ssh first and then kv, you won't see the KV options in the mount form. 
![with-cached-model-value](https://user-images.githubusercontent.com/82459713/227601014-665a78df-b93a-44d0-b43f-bad8a27f0ace.gif)

Reply:
Update: I've refactored such that the forms still update, and we're not calling `expandAttributeMeta` between renders!